src/theory/arith/nl/ext/proof_checker.cpp

```diff
@@ -45,7 +48,123 @@ Node ExtProofRuleChecker::checkInternal(PfRule id,
   {
     Trace("nl-ext-checker") << "\t" << c << std::endl;
   }
-  if (id == PfRule::ARITH_MULT_TANGENT)
+  if (id == PfRule::ARITH_MULT_SIGN)
+  {
+    Assert(children.empty());
+    Assert(args.size() > 1);
+    Node mon = args.back();
+    std::map<Node, int> exps;
+    std::vector<Node> premise = args;
+    premise.pop_back();
+    Assert(mon.getKind() == Kind::MULT
+           || mon.getKind() == Kind::NONLINEAR_MULT);
+    for (const auto& v : mon)
+    {
+      auto it = exps.find(v);
+      if (it == exps.end())
+      {
+        exps.emplace(v, 1);
+      }
+      else
+      {
+        ++it->second;
+      }
```

Comment:
Fun fact about `std::map`: you can access missing entries using `operator[]` and it default-initializes (i.e. equivalent to `{}`-initilizing) the value for that key if there is no value there already.

ergo, `exps[v]++` suffices here.

Perhaps this shouldn't be used though... I'm not sure whether most C++ devs know this is how the language works.